core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java

```diff
@@ -1854,7 +1854,7 @@ public class SqlStdOperatorTable extends ReflectiveSqlOperatorTable {
   public static final SqlFunction PI =
       SqlBasicFunction.create("PI", ReturnTypes.DOUBLE, OperandTypes.NILADIC,
           SqlFunctionCategory.NUMERIC)
-          .withSyntax(SqlSyntax.FUNCTION_ID);
+          .withSyntax(SqlSyntax.FUNCTION_ID_CONSTANT);
```

Comment:
aren't there more functions that need this syntax?
I expect many system functions would

Reply:
This type of function is special because of the current implementation of Calcite. In fact, this type of function requires parentheses to be called in most databases (I haven't found any other databases yet).

Reply:
what about all the system functions like USER, CURRENT_TIME, or functions like RANDOM?

Reply:
The Function like USER、CURRENT_TIME、CURRENT_USER、SESSION_USER、SYSTEM_USER、CURRENT_PATH、CURRENT_ROLE、CURRENT_CATALOG、CURRENT_SCHEMA、LOCALTIME、LOCALTIMESTAMP、CURRENT_TIME and CURRENT_TIMESTAMP behaviour controlled by `allowNiladicParentheses`. Before this PR, The PI is also controlled by `allowNiladicParentheses`, but this will result in differences in behavior compared to other databases. In other databases the PI always needs parentheses. I add `allowNiladicConstantWithoutParentheses`.On the one hand, it is to be compatible with the current implementation of Calcite, and on the other hand, to make its calls more in line with the usage of most databases.

Reply:
Yes. The PI behavior is different from the User. The PI always needs to use PI(). Before this PR, Calcite treats PI and USER as a class of functions.

Reply:
I think this is an important point, and it would be nice to have this in the code somewhere as a comment.